### ISSUE TITLE
Expose the name of the current toolbox container

### ISSUE THREAD

**Issue by pierreprinetti** [containers/toolbox#744]
**Is your feature request related to a problem? Please describe.**
I want to display the name of the toolbox I'm currently in in my prompt, or in my tmux bar, without relying on the container engine underlying Toolbox.

**Describe the solution you'd like**
While there is [an open issue](https://github.com/containers/toolbox/issues/576) specifically targeting the stock Bash `$PS1`, I'd fancy a backend-agnostic `toolbox` command that just outputs the current toolbox name. The reason for this is: I use a customised command prompt and I use ZSH.

Here's a proposal:

```shell
$ toolbox show
fedora-toolbox-33
```

Basically, `toolbox show` outputs the value of `/run/.containerenv`'s `name`.

The command exits with 1 if the current shell is not running in a toolbox.

**Describe alternatives you've considered**

* `toolbox list --current`
* `toolbox show <toolbox>` outputs some interesting info on the container in a machine-readable format, and defaults to the current toolbox (errors if not in a toolbox, and `<toolbox>` is not passed)
* I can continue parsing `/run/.containerenv`.

**Additional context**
Depends on https://github.com/containers/podman/issues/6192

**Comment by erindru** [comment]
In case anyone else stumbles across this looking for a copy+paste solution before this becomes a proper feature in Toolbox, here is what I added to my `.bashrc` to parse the container name out of `/run/.containerenv`:

```sh
if [ -f "/run/.containerenv" ]; then
	TOOLBOX_NAME=[$(cat /run/.containerenv | grep 'name=' | sed -e 's/^name="\(.*\)"$/\1/')]
else
	TOOLBOX_NAME=""
fi
```

It can then be used within the `$PS1` to produce something like:

![Screenshot from 2021-04-20 08-56-39](https://user-images.githubusercontent.com/1001316/115302334-70970f80-a1b6-11eb-9878-3ee2d1c558ea.png)

**Comment by ma3yta** [comment]
My [starship](https://starship.rs/) config to get container name.

```ini
format = """${custom.toolbox}$directory$git_branch$git_metrics$git_status$dotnet$character"""

[custom.toolbox]
command = """ grep -Po 'name="\\K[^"]+' /run/.containerenv """
files = [ "/run/.containerenv" ]
when = "test -f /run/.containerenv"
style = "bg:bright-blue black"
format = "[  $output ]($style)"
```
![Screenshot from 2021-09-13 15-42-10](https://user-images.githubusercontent.com/660379/133052531-b3359059-14f1-4df2-91f2-78029a9e31fd.png)

**Comment by pierreprinetti** [comment]
I ended up putting this in `~/.local/bin/get_toolbox`:

```bash
#!/usr/bin/env bash

set -Eeuo pipefail

declare -r containerenvpath='/run/.containerenv'

if [ -r $containerenvpath ]; then
        source $containerenvpath
        printf '%s' "$name"
else
        exit 1
fi
```

**Comment by debarshiray** [comment]
Oops!  I have no idea how I managed to miss this issue for so long.

> I can continue parsing /run/.containerenv.

Yes, please do continue to parse `/run/.containerenv`.

In https://github.com/containers/podman/issues/6192 I had requested for the metadata that shows up in `/run/.containerenv` specifically for such use-cases, and I consider the contents of that file as part of our stable interface.  So, please do file issues against [Podman](https://github.com/containers/podman) if that interface breaks, and CC me, and we will work out a fix.

The format of `/run/.containerenv` matches that of [os-release(5)](https://www.freedesktop.org/software/systemd/man/os-release.html) and is similar to Flatpak's `/.flatpak-info`.  So it's as good an interface as any.

You can also get the same metadata through `podman inspect ...`, but in a different format.

I don't want to add yet another way to access that metadata to Toolbx, because it will be a duplicated interface that we will have to maintain, write tests that get executed on every CI run, etc..  Instead let's just rely on the existing mechanisms.

That said, if the existing mechanisms don't work for some use-case, then please do file a new issue with details.

**Comment by debarshiray** [comment]
Closing.

Thanks for playing with Toolbx and sharing those interesting snippets.  :)
